tock/hours/views.py

```diff
@@ -313,6 +315,64 @@ def form_valid(self, form):
     def get_success_url(self):
         return reverse("ListReportingPeriods")
 
+def display_add_hours_error_message(request):
+    error_msg = "Oops. That command was not correct and no time was added to your timecard. Try again by entering a URL with this format: tock.gov/addHours?project=231&hours=1"
+    messages.add_message(
+        request,
+        messages.ERROR,
+        error_msg
+    )
+
+
+def add_hours_view(request):
+    project_id = request.GET['project']
+    hours = Decimal(request.GET['hours'])
+    r = ReportingPeriod.objects.latest()
+
+    try:
+        proj = Project.objects.get(id=project_id)
+    except Project.DoesNotExist:
+        display_add_hours_error_message(request)
+        return redirect(reverse('reportingperiod:UpdateTimesheet', args=[r]))
+
+    tc, created = Timecard.objects.get_or_create(
+        reporting_period_id=r.id,
+        user_id=request.user.id)
+
+    if created:
+        tc.save()
+
+    tco, created = TimecardObject.objects.get_or_create(
+        timecard_id=tc.id,
+        project_id=proj.id)
+
+    if tco.hours_spent is None:
+        tco.hours_spent = 0
+        display_add_hours_error_message(request)
```

Comment:
@elg0nz When new TimecardObject's are initialized, they have a hours_spent value of None. Redirecting a user with an error message if a TimecardObject's hours_spent attribute is None means that the success of a URL-based addHours request is dependent on whether the user has: (1) an existing timecard for the period; and (2) has a TimecardObject with a project attribute that matches the project param of the request. Otherwise, the code will (correctly) initialize a new TimecardObject with hours_spent as None and promptly redirect with an error message.